src/Aspire.Hosting/ApplicationModel/EndpointReference.cs

```diff
@@ -48,6 +48,11 @@ public sealed class EndpointReference : IManifestExpressionProvider, IValueProvi
     /// </summary>
     internal string GetExpression(EndpointProperty property = EndpointProperty.Url)
     {
+        if (property == EndpointProperty.HostAndPort)
+        {
+            return $"{{{Resource.Name}.bindings.{EndpointName}.host}}:{{{Resource.Name}.bindings.{EndpointName}.port}}";
```

Comment:
I'm not hugely familiar with the manifest side of aspire as I've not used it for deployment.  Faking `HostAndPort` through separate host and port bindings works, but I'm not sure if it would be better to introduce a `hostandport` binding into the manifest format.

Reply:
Why isn't ths in the switch? BTW this is the right approach.